### PR TITLE
MANT: Use npy_cache_import for all _internal.py imports

### PR DESCRIPTION
**Pull request by jaimefrio**

### CODE REVIEW

**Comment by juliantaylor** [comment]
isn't this already cached inside python?

**Comment by jaimefrio** [comment]
That would explain why there is no performance improvement in the sort-like functions using the `order` kwarg with or without this caching.

I was mostly doing this for consistency of the method used. And it makes returning values a tad more straightforward from some functions, as you don't need to explicitly decref the imported module. But it also requires that you scroll up to see what it was that `importfunc` was when looking at the code for the first time, which is not the best one could wish for.

It certainly looked much better in my mind that what it has turned out to be, so feel free to close it if you think the benefits are not there.

**Comment by seberg** [comment]
Hmm, if there is no performance gain, we might want to do the opposite even. The cache import does leak references. I am personally not that afraid about leaking references on the module level, but if there is no gain, maybe we should not do this.

**Comment by jaimefrio** [comment]
The nice thing to do then would be to, instead of `npy_cache_import`, have a single function, `npy_import_and_call`, taking the module name string, the function name string, the parameters string, and the variable number of parameters and returning the result of the call.

**Comment by charris** [comment]
@jaimefrio Should we keep this open?

**Comment by mattip** [comment]
Can we close this?

File: numpy/core/src/multiarray/descriptor.c [hunk 1]
```diff
@@ -698,17 +698,16 @@ _convert_from_commastring(PyObject *obj, int align)
...
 
-    if (!PyBytes_Check(obj)) {
+    npy_cache_import("numpy.core._internal", "_commastring", &importfunc);
+    if (importfunc == NULL) {
         return NULL;
     }
-    _numpy_internal = PyImport_ImportModule("numpy.core._internal");
-    if (_numpy_internal == NULL) {
+    if (!PyBytes_Check(obj)) {
         return NULL;
```
Comment (eric-wieser) [inline numpy/core/src/multiarray/descriptor.c]: This returns null without setting an exception

**Comment by mattip** [comment]
Closing, please reopen or open a new PR if there are new insights about caching

**Pull request closed without merging**